chaosazure/vmss/fetcher.py

```diff
@@ -26,62 +41,54 @@ def choose_vmss_instance_at_random(vmss_choice, configuration, secrets):
 def choose_vmss_instance(vmss_choice: dict,
                          configuration: Configuration = None,
                          instance_criteria: Iterable[Mapping[str, any]] = None,
-                         secrets: Secrets = None) -> Dict[str, Any]:
-    vmss_instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
-    if not vmss_instances:
-        logger.debug("No virtual machine scale set instances found")
-        raise FailedActivity("No virtual machine scale set instances found")
-    else:
-        logger.debug(
-            "Found virtual machine scale set instances: {}".format(
-                [x['name'] for x in vmss_instances]))
-
-    choice_vmss_instance = None
-    for vmss in vmss_instances:
-        if vmss_matches_criteria(vmss, instance_criteria):
-            choice_vmss_instance = vmss
-            break
+                         secrets: Secrets = None) -> List[Dict[str, Any]]:
+    result = []
+    instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
 
-    if not choice_vmss_instance:
-        logger.debug("No virtual machine scale set instance found for "
-                     "virtual machine scale set %s and criteria %s"
-                     % (vmss, instance_criteria))
-        raise FailedActivity("No virtual machine scale set instances found for"
-                             " criteria")
+    for instance in instances:
+        if is_criteria_matched(instance, instance_criteria):
+            result.append(instance)
 
-    logger.warning("Attempting to stop instance with name %s"
-                   % choice_vmss_instance['name'])
+    if len(result) == 0:
+        msg = "No VMSS instance found for criteria '{}'".format(
+            instance_criteria)
+        logger.debug(msg)
+        raise FailedActivity(msg)
```

Comment:
Do we not log an error in case FailedActivity? if yes please remove debug massege as it is redundant